engine/src/main/java/org/camunda/bpm/engine/impl/batch/BatchEntity.java

```diff
@@ -368,6 +374,15 @@ public void delete(boolean cascadeToHistory, boolean deleteJobs) {
     }
   }
 
+  protected void deleteVariables(CommandContext commandContext) {
+    VariableInstanceManager variableInstanceManager = commandContext.getVariableInstanceManager();
+
+    List<VariableInstanceEntity> variableInstances =
+        variableInstanceManager.selectByBatchId(id);
+
+    variableInstances.forEach(VariableInstanceEntity::delete);
+  }
+
```

Comment:
I already thought about this but discarded the idea since one delete statement is not enough. We need to delete the variable instances *AND* the associated byte arrays. While the variable instance delete statement would be relatively simple, the delete statement for the byte arrays needs to subselect over the variables table. Some databases (like MySQL) might perform poorly with subselects, and the performance improvement we would gain for this specific use case is probably neglectable.

Reply:
Agree 👍 Let's keep it as-is then. Thanks for the input!